website/content/docs/enterprise/consistency.mdx

```diff
@@ -0,0 +1,212 @@
+
+## Performance replication
+
+A similar phenomenon occurs when using performance replication.  One example
+of how this manifests is when using shared mounts.  If a KV secrets engine
+is mounted on the primary with `local=false`, it will appear on the secondary
+cluster as well.  The secondary cluster can handle requests to that mount,
+though as with performance standbys, write requests must be forwarded - in
+this case to the primary active node.  Once again, if a write request is followed
+by a read request, and the data initially written on the primary cluster active
+node hasn't yet been pushed out to the secondary cluster before the read request is
+processed on the secondary node handling the request, it will appear as though
+the write hasn't happened.  If the secondary cluster is using Integrated Storage,
+and the read request is being handled on a performance standby, the problem is
+exacerbated because it has to be sent first from the primary active node to the
+secondary active node, and then from there to the secondary performance standby.
+
+Even without shared secret engines this can still happen with performance
+replication. The Identity subsystem aims to provide a view on entities and
+groups which spans clusters.  As such, when logging in to a secondary cluster
+using a shared mount Vault tries to generate an entity and alias if they don't
+already exist, and these must be stored on the primary using an RPC.  If the
+auth method defines external groups, the login process will also try to refresh
+those so that the resulting token reflects the entity's group memberships.
+This also applies to token renew operations.
+
+(TODO: remove prev paragraph?  maybe more background than needed)
+
+## Mitigations
+
+There has long been a partial mitigation for the above problems.  When writing
+data via RPC, e.g. when a performance standby registers tokens and leases on the
+active node after a login or generating a dynamic secret, part of the response
+includes a number known as the "WAL index".  A full explanation of this is
+outside the scope of this document, but suffice it to say that both performance
+replication and performance standbys use something called log shipping to stay
+in sync with the upstream source of writes.  The mitigation used by nodes
+sending writes upstream via RPC is to look at the WAL index in the response
+and wait up to 2 seconds to see the that WAL index appear in the logs being
+shipped from upstream.  If the WAL index is seen, the Vault node handling the
+request that resulted in RPCs can return: it knows that any subsequent reads
+will be able to see the value that was just written.  If the WAL index isn't
+seen within those 2 seconds, the Vault node completes the request and hopes
+for the best.
+
+This mitigation option still exists in Vault 1.7, though now there is a
+configuration option to adjust the wait time.
+
+(TODO: reference new config option)
```

Comment:
Just flagging this should get replaced with something else.

Reply:
Done.